sdk/eventhub/event-hubs/samples/eventProcessor.ts

```diff
@@ -1,25 +1,42 @@
 import {
   EventHubClient,
-  EventData,
+  ReceivedEventData,
   EventPosition,
   delay,
   EventProcessor,
-  PartitionContext
+  PartitionContext,
+  InMemoryPartitionManager,
+  CheckpointManager
 } from "@azure/event-hubs";
 
 class SimplePartitionProcessor {
   private _context: PartitionContext;
-  constructor(context: PartitionContext) {
+  private _checkpointManager: CheckpointManager;
+  constructor(context: PartitionContext, checkpointManager: CheckpointManager) {
     this._context = context;
+    this._checkpointManager = checkpointManager;
   }
-  async processEvents(events: EventData[]) {
+  async processEvents(events: ReceivedEventData[]) {
     for (const event of events) {
```

Comment:
There's a chance that `events` might be an empty array.  For this sample, can you check the length and return right away if it's 0?  Otherwise the `updateCheckpoint` below could throw an error.